### PR TITLE
Support full observation error matrix

### PR DESCRIPTION
**Pull request by dafeda**
Resolves: #64

### CODE REVIEW

**Review by frode-aarstad — APPROVED** [review]
R is no more. LGTM

File: tests/test_theoretical_property.py [hunk 1]
```diff
@@ -20,7 +32,7 @@ def test_that_update_is_according_to_theory():
     """
     N = 1500
     nparam = 3
-    var = 2
+    var = 2.0
```
Comment (ManInFez) [inline tests/test_theoretical_property.py:35]: why...

Comment (dafeda) [reply tests/test_theoretical_property.py]: Just checking that you are paying attention.
Fixed 👍

File: src/iterative_ensemble_smoother/ies.cpp [hunk 2]
```diff
@@ -287,7 +274,7 @@ create_transition_matrix(py::EigenDRef<MatrixXd> Y, py::EigenDRef<MatrixXd> R,
   if (ies_inversion == Inversion::exact) {
     exact_inversion(W, S, H, ies_steplength);
   } else {
-    subspace_inversion(W, ies_inversion, E, R, S, H, truncation,
+    subspace_inversion(W, ies_inversion, E, R.value(), S, H, truncation,
```
Comment (ManInFez) [inline src/iterative_ensemble_smoother/ies.cpp:277]: Won`t `R.value()` raise a `[std::bad_optional_access](https://en.cppreference.com/w/cpp/utility/optional/bad_optional_access)` if R is not an actual matrix?

Comment (dafeda) [reply src/iterative_ensemble_smoother/ies.cpp]: Good question, thanks.
I've now made `R` optional in `subspace_inversion` and made some changes to how `R` is created in the first place.
I've also added tests to make sure every inversion method works when `observation_errors` are either a 1D array or a 2D array.

File: tests/test_theoretical_property.py [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+var = 2.0
 
-def test_that_update_is_according_to_theory():
+
+@pytest.mark.parametrize(
+    "inversion,errors",
+    [
+        (ies.InversionType.EXACT, np.diag(np.array([var, var, var]))),
+        (ies.InversionType.EXACT, np.array([np.sqrt(var), np.sqrt(var), np.sqrt(var)])),
+        (ies.InversionType.EE_R, np.diag(np.array([var, var, var]))),
```
Comment (ManInFez) [inline tests/test_theoretical_property.py]: missing EE_R for np.array

**Review by ManInFez — APPROVED** [review]
:bacon:

**Pull request merged**